webhooks/cluster.go

```diff
@@ -103,3 +143,14 @@ func (c *Cluster) ValidateDelete(ctx context.Context, obj runtime.Object) error
 	}
 	return cluster.ValidateDelete()
 }
+
+// checkClusterClassExists checks for the passed clusterClass in the API server.
+func (c *Cluster) checkClusterClassExists(ctx context.Context, namespace, clusterClass string) *field.Error {
+	if err := c.Client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: clusterClass}, &clusterv1.ClusterClass{}); err != nil {
+		return field.Invalid(
+			field.NewPath("spec", "topology", "class"),
+			clusterClass,
+			"clusterv1.ClusterClass could not be found")
+	}
+	return nil
```

Comment:
nit: up to you if you already want to consider it. Absolutely fine to ignore it and discuss/change it in later PRs.

We will need the clusterClass in later PRs, as we don't want to do multiple get's it might make sense to already return the ClusterClass here.

Reply:
I think I'll do this in a follow up PR.

Reply:
Makes sense!